pkg/clients/edgeconnect/environment_settings.go

```diff
@@ -69,7 +69,7 @@ func (c *client) GetConnectionSetting(uid string) (EnvironmentSetting, error) {
 	}
 
 	for _, item := range resDataJson.Items {
-		if item.Value.UID == uid {
+		if item.Value.Name == name && item.Value.Namespace == namespace && item.Value.UID == uid {
```

Comment:
could you please add a test, which checks that we mismatch `name` or `namespace` like written in ticket?
As you can see we never return empty:
<img width="483" alt="image" src="https://github.com/Dynatrace/dynatrace-operator/assets/1374633/b1770c51-5326-403a-91dd-ee05b1ee901f">

Reply:
Done [here](https://github.com/Dynatrace/dynatrace-operator/pull/3359/commits/d3803495de1b8596a2e998b139c62675ef5b4131)
I also refactored a bit the implementation, removing logic from the client into the controller (it was also necessary to test the logic)